src/platform/Darwin/DnssdInterfaceInfos.h

```diff
@@ -0,0 +1,77 @@
+/*
+ *
+ *    Copyright (c) 2022 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <dns_sd.h>
+#include <lib/dnssd/platform/Dnssd.h>
+
+#include <string>
+#include <vector>
+
+namespace chip {
+namespace Dnssd {
+
+struct RegisterContext;
+struct RegisterRecordContext;
+
+class InterfaceInfos
+{
+public:
+    bool Init(const struct ifaddrs * ifa, Inet::IPAddressType addressType, uint32_t interfaceId);
+    DNSServiceErrorType Register(DNSServiceRef sdRef, RegisterRecordContext * sdCtx, const char * hostname) const;
+
+private:
+    bool HasValidFlags(unsigned int flags) const;
+    bool HasValidType(Inet::IPAddressType addressType, const sa_family_t addressFamily) const;
+    bool IsValidInterfaceId(uint32_t targetInterfaceId, unsigned int currentInterfaceId) const;
+
+    const struct in_addr * GetIPv4() const { return &mInterfaceAddress.ipv4; }
+    const struct in6_addr * GetIPv6() const { return &mInterfaceAddress.ipv6; }
+
+    bool IsIPv4() const { return mInterfaceAddressType == AF_INET; };
+    bool IsIPv6() const { return mInterfaceAddressType == AF_INET6; };
+
+    void LogDetail() const;
+
+    union InterfaceAddress
+    {
+        struct in_addr ipv4;
+        struct in6_addr ipv6;
+    };
+
+    uint32_t mInterfaceId;
+    InterfaceAddress mInterfaceAddress;
+    sa_family_t mInterfaceAddressType;
+};
+
+class InterfacesInfos
```

Comment:
This and `class InterfaceInfos` are _very_ easy to mistake for each other.

Can we come up with better naming here, please?  What does an `InterfacesInfos` actually represent?  What does an `InterfaceInfos` actually represent?